snappy_wrappers/wrappers/somatic_variants_checking/summarize-vcf.py

```diff
@@ -0,0 +1,254 @@
+
+
+def check_variant_in_bed(v_chrom, v_start, v_end, bed_intervals, padding=0):
+    try:
+        tmp = []
+        records = bed_intervals.query(v_chrom, v_start - padding, v_end + padding)
+        for record in records:
+            tmp.append(record)
+        if len(tmp) >= 1:
+            return True
+        else:
+            return False
+    except OSError:
+        print("You should index the bed file with tabix first!")
+
+
+def get_contigs_from_bed_file(bedfile):
+    contigs = []
+    try:
+        with gzip.open(bedfile, "r") as intervals:
+            for interval in intervals:
+                line = interval.decode()
+                tmp = line.split("\t")[0]
+                if tmp not in contigs:
+                    contigs.append(tmp)
+        return contigs
+    except OSError:
+        print("Error reading the GZIP file.")
+
+
+def get_variant_type(ref, alt):
+    if (len(alt) == 1) and (len(ref) == len(alt)):
+        return "snp"
+    elif len(alt) != len(ref):
+        return "indels"
+
+
+def check_sp_read(variant, minimal, limited):
+    dp = variant.INFO["DP"]
+    if dp <= minimal:
+        return "minimal"
+    elif (dp > minimal) and (dp <= limited):
+        return "limited"
+    elif dp >= limited:
+        return "strong"
+
+
+def assign_class_snvs(variant, mt_mat):
+    temp = str(variant.REF) + ">" + str(variant.ALT[0])
+    # Following result from plot-vcfstats with bcftools
+    if temp in ["A>C", "T>G"]:
+        mt_mat[0] += 1
+    elif temp in ["A>G", "T>C"]:
+        mt_mat[1] += 1
+    elif temp in ["A>T", "T>A"]:
+        mt_mat[2] += 1
+    elif temp in ["C>G", "G>C"]:
+        mt_mat[3] += 1
+    elif temp in ["C>T", "G>A"]:
+        mt_mat[4] += 1
+    elif temp in ["G>T", "C>A"]:
+        mt_mat[5] += 1
+    return mt_mat
+
+
+def process_vcf_file(
+    vcf_file,
+    contigs=[],
+    hard_contigs=[],
+    minimal=1,
+    limited=5,
+    bed_file="",
+    hard_file="",
+    filter_file=False,
+    padding=0,
+):
+    infor = {
+        "total_v_count": 0,
+        "v_inside_exom": 0,
+        "v_outside_exom": 0,
+        "v_in_hard_regions": 0,
+        "n_snps": 0,
+        "n_indels": 0,
+        "minimal_rp_snvs_exom": 0,
+        "minimal_rp_snvs_nexom": 0,
+        "limited_rp_snvs_exom": 0,
+        "limited_rp_snvs_nexom": 0,
+        "strong_rp_snvs_exom": 0,
+        "strong_rp_snvs_nexom": 0,
+        "indels_length": [],
+        "mt_classes": [0] * 6,
+        "vaf": [],
+    }
+    if filter_file:
+        for variant in vcf_file:
+            infor["total_v_count"] += 1
+            # Gathering information of variants in comparison to interested regions
+            if variant.CHROM in contigs:
+                if len(variant.ALT) > 1:
+                    raise ValueError(
+                        "Multiallelic variants are not supported"
+                    )  # checking for multiallelic
+                infor["vaf"].append(float(variant.format("AF")[1][0]))
```

Comment:
`AF` is not a reserved field in the vcf 4.3 format. Other callers may use a different id. You should have the ID under user control (with `AF` as a default).